### ISSUE TITLE
v2.19.0: Vaccination status tag 2G+ gone

### ISSUE THREAD

**Issue by NKarolak** [corona-warn-app/cwa-app-android#4959]
## Avoid duplicates
* [x] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [x] Bug is specific for Android only, for general issues / questions that apply to iOS and Android please raise them in the [documentation repository](https://github.com/corona-warn-app/cwa-documentation)
* [x] Bug is not already reported in another issue

## Technical details

- Device name: Samsung Galaxy s10e
- Android version: 12
- App version: 2.19.0

## Describe the bug

I have 3 vaccination certificates:
![image](https://user-images.githubusercontent.com/34504100/159015279-9f311ab4-b734-4731-871f-551b0b34d092.png)
All certificates have been created for the very same name. They were (and still are) grouped. There is no second certification group.

Before the update to v2.19, my certificate had the "2G+" tag.
Since the update, there is no tag at all:
![image](https://user-images.githubusercontent.com/34504100/159015488-4c9cd7b2-c076-4a1d-853e-3b06bd2d41f4.png)

App restart and phone restart did not help.

## Steps to reproduce the issue

1. Go to 'Certificates'
2. See that the tag is entirely missing.

## Expected behaviour

Show a tag for 2G+.

**Comment by Ein-Tim** [comment]
Did you try removing one of the certificates (e.g. the 3/3 one) and readding it? If not, could you please try this and let us know if this changed something.

Still this bug should be investigated, not sure if an error log would help here @mlenkeit?

**Comment by dsarkar** [comment]
FYI - we have reproduced this bug

**Comment by NKarolak** [comment]
@dsarkar Thanks, deleting the test certificate solved it!

**Comment by dsarkar** [comment]
@NKarolak Thanks for the feedback. So, this is the workaround for the moment, however we have an internal bug ticket: 
Internal Tracking ID: [EXPOSUREAPP-12365](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-12365)

Best wishes, DS

---
Corona-Warn-App Open Source Team

**Comment by mlenkeit** [comment]
@NKarolak thanks for creating this report! Do you recall what you did between last time you saw 2G+ and the time it did not show? Did you add any new certificates?  Did you upgrade to 2.19? (I saw you're on 2.19 now, but wondering if it already happened on 2.18)

**Comment by NKarolak** [comment]
@mlenkeit I am not sure when exactly was the last time I saw 2G+.
Within the last week, I did:
- Enter new locations into the diary
- I scanned a check-in (after that my certificate was checked by the staff, but I did not pay enough attention to remember whether the flag was already gone or not). This was 6 days ago.
- Upgrade to 2.19 (yesterday).

However, I am sure that earlier under 2.18, 2G+ was shown.

**Comment by mlenkeit** [comment]
@NKarolak you mentioned in https://github.com/corona-warn-app/cwa-documentation/issues/871#issuecomment-1072473847 that all your certificates have the same name. Are you 100% sure that before 2.19, they were all grouped under the same person? Slight name differences such as middle name or so might already be sufficient.

**Comment by NKarolak** [comment]
Yes, by the time I updated, they were all grouped under the same person. Again: there is and was no second group.

I am very sure because I took care to reach that state ;-)
Much earlier, my third certificate was missing my middle name, hence I had two groups. That annoyed me, so that I got another certificate with the correct name, deleted the old certificate, and imported the corrected one.
When I checked today, there was nothing left in the trash can from that exchange.

**Comment by mtwalli** [comment]
@NKarolak, Sorry for bothering you with many questions ,but to better assess the issue. I have one more question 🙋‍♂️. does your name have middles names? not just `firstName, lastName`?

**Comment by NKarolak** [comment]
@mtwalli Alright ;-)
Like I said:
>  was missing my middle name

So yes, my certificates consist of 3 names.

**Comment by mtwalli** [comment]
@NKarolak, No worries 😉, I'm so sorry that you are experiencing such issue, It would be helpful to send us a log report ,since you can reproduce it by restoring the certificate back in the wallet ?

**Comment by NKarolak** [comment]
Hopefully I understood the process of sending the logs correctly ...

So, here is the ID: 2531E34105C1360BFB49.

Given: 
The test certificate is in the recycle bin, hence I am seeing the status proof line 2G+.

Performed steps in the log:

1. Restored the test certifcate, and opened the certifcates: status proof line gone.
2. Deleted the test certificate, and opened the certificates: status proof line restored.

**Comment by Ein-Tim** [comment]
@mlenkeit I have an affected user on Twitter who has only stored their 3 vaccination certificates. What is the suggested workaround for them? Completely deleting the certificate and re-adding it? Will this help?

**Comment by Ein-Tim** [comment]
Just for the record I can also reproduce this issue.

**Comment by mtwalli** [comment]
> Just for the record I can also reproduce this issue.

can you share the test certificates you used to reproduce?

**Comment by Ein-Tim** [comment]
@mtwalli 

I have no test certificates but only a 2/2 vac cert and a 3/3 vac cert which were grouped together in 2.19.

**Comment by ZarVladimirII** [comment]
Opening it today, I got confused. I had explicitly rapid-test and vaccination divided (and happy about it) and the 2G/3G signs are gone.
Also middlename used for vaccination certificate.

**Comment by MikeMcC399** [comment]
- There is a solution for this issue being prepared under https://github.com/corona-warn-app/cwa-app-android/pull/4962

**Comment by Necr1s** [comment]
The issue still persists for me in Version 2.19.1. I have three vaccination vertificates with standardised name `LASTNAME<<PRENAME` and one recovered certificate ( if that is the correct term ) with the standadised name `LASTNAME<<PRENAME<MIDDLENAME`. Deleting the recovered certificate solves this issue, but it will be back if I restore it from the bin.

**Comment by Ein-Tim** [comment]
There is another PR for this, it will be part of the 2.19.2 release: https://github.com/corona-warn-app/cwa-app-android/pull/4969

**Comment by MikeMcC399** [comment]
- PR https://github.com/corona-warn-app/cwa-app-android/pull/4962 went into pre-release version [v2.19.2-rc.0](https://github.com/corona-warn-app/cwa-app-android/releases/tag/v2.19.2-rc.0)
- PR https://github.com/corona-warn-app/cwa-app-android/pull/4969 went into pre-release version [v2.19.2-rc.1](https://github.com/corona-warn-app/cwa-app-android/releases/tag/v2.19.2-rc.1)

**Comment by mtwalli** [comment]
> The issue still persists for me in Version 2.19.1. I have three vaccination vertificates with standardised name `LASTNAME<<PRENAME` and one recovered certificate ( if that is the correct term ) with the standadised name `LASTNAME<<PRENAME<MIDDLENAME`. Deleting the recovered certificate solves this issue, but it will be back if I restore it from the bin.

The fix is planned in `2.19.2` (Another hotfix)

**Comment by Ein-Tim** [comment]
Version 2.19.2 is now released here on GitHub: https://github.com/corona-warn-app/cwa-app-android/releases/tag/v2.19.2

**Comment by mtwalli** [comment]
Yes, released to 10% now

**Comment by mtwalli** [comment]
Released to 💯% , probably a few hours until general availability

**Comment by mtwalli** [comment]
@MikeMcC399 ,Good!, so users who are affected by the issue could finally update and let us know their feedback

**Comment by NKarolak** [comment]
I've just updated to 2.19.2, and I can confirm that the issue is now solved for me: when I restore the deleted test certificate, there is still only one certificate group, and it does have the status line.

**Comment by Necr1s** [comment]
I can confirm that too. It has worked even with a non deleted certificate ( besides the deletion and restoration yesterday ). Thanks for fixing it!

**Comment by ZarVladimirII** [comment]
Everything works again. Status, switching test and vaccination, days since last vaccination(was also gone).

**Comment by dsarkar** [comment]
@NKarolak @Necr1s @ZarVladimirII Thanks for the feedback and confirming the fix. Best wishes, DS

---
Corona-Warn-App Open Source Team
